src/kernels/errors/kernelErrorHandler.ts

```diff
@@ -306,6 +321,7 @@ export abstract class DataScienceErrorHandler implements IDataScienceErrorHandle
             }
             return KernelInterpreterDependencyResponse.cancel;
         } else if (err instanceof JupyterSelfCertsError) {
+            this.sendKernelTelemetry(err, errorContext, resource, err.category);
```

Comment:
Kernel Telemetry (for failrues) is sent in this class after we've analysed the failure.